### ISSUE TITLE
Unable to assign static IP to Ubuntu 22.04 image result with cloud-init

### ISSUE THREAD

**Issue by a-belhadj** [vmware-samples/packer-examples-for-vsphere#981]
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.

### Project Version

latest

### VMware vSphere

8.0.3.00100

### HashiCorp Packer

1.11.2

### HashiCorp Packer Plugin for VMware vSphere

v1.4.0

### Guest Operating System

Ubuntu Server 22.04

### Environment Details

Template deployed with Ansible, the same cloud-init (userdata and metadata) works with RHEL and Rocky template

### Description

Hello,

When using this project to build an Ubuntu 22.04 or 24.04 VM template with Packer and Cloud-Init enabled, I am unable to set a static IP when instantiating the VM. Instead, the VM consistently uses the IP that was assigned during the build process.

### Expected Behavior

Be able to inject static configuration IP via metadata. 
The current workaround is the following
```yaml
#cloud-config
    timezone: Europe/Paris
    users:
      - name: cloud-user
        ssh_authorized_keys: {{ ssh_authorized_keys | to_json }}

    write_files:
    - path: /etc/cloud/cloud.cfg.d/90-network.cfg
      owner: root:root
      permissions: 0o600
      defer: true
      content: |
        network:
          version: 2
          ethernets:
            ens192:
              dhcp4: false
              dhcp6: false
              addresses:
                - "{{ ip_address }}"
              gateway4: "{{ gateway }}"
              nameservers:
                addresses: {{ dns }}
    runcmd:
      - rm /etc/netplan/50-cloud-init.yaml
      - rm /etc/cloud/cloud.cfg.d/90-installer-network.cfg
      - cloud-init clean --reboot
```

### Actual Behavior

After creating the image with Packer and instantiating the template in VMware:
1. The VM boots with the network IP set during the build process.
2. Any attempt to inject a static IP configuration using Cloud-Init is ignored.
3. The following files contain traces of the network configuration from the build:
   - `/etc/netplan/50-cloud-init.yaml` contains the original network configuration from the build.
   - `/etc/cloud/cloud.cfg.d/90-installer-network.cfg` also retains traces of the initial build network settings.
4. Good specs are available in: 
   - `/run/cloud-init/instance-data.json` but it does not seem to propagate properly to Cloud-init and then Netplan.

### Steps to Reproduce

Build with cloud_init_enable_true and deploy in VMWare with:
```yaml
advanced_settings:
      - key: "guestinfo.metadata"
        value: "{{ cloud_init.metadata | b64encode }}"
      - key: "guestinfo.metadata.encoding"
        value: "base64"
      - key: "guestinfo.userdata"
        value: "{{ cloud_init.userdata | b64encode }}"
      - key: "guestinfo.userdata.encoding"
        value: "base64"
```
metadata file
```yaml

local-hostname: "vm_name"
instance-id: "vm_name"
network:
  version: 2
  ethernets:
    ens192:
      dhcp4: false
      dhcp6: false
      addresses:
        - "x.x.x.24/20"
      gateway4: "x.x.x.1"
      nameservers:
        addresses:
          - x.x.x.x
```

I also tested with network.version 1 and it's the same

### Log Fragments and Files
Logs from inside the new VM, I connect to it with the build IP
``` bash
cloud-user@vm:~$ sudo cloud-init schema --system 
Found cloud-config data types: user-data, network-config

1. user-data at /var/lib/cloud/instances/156d2742-3d2b-4b18-fb77-9419a4afb558/cloud-config.txt:
  Valid schema user-data

2. network-config at /var/lib/cloud/instances/156d2742-3d2b-4b18-fb77-9419a4afb558/network-config.json:
Skipping network-config schema validation for version: 2. No netplan API available.

cloud-user@vm:~$ sudo cat /var/lib/cloud/instances/156d2742-3d2b-4b18-fb77-9419a4afb558/network-config.json 
{
 "ethernets": {
  "ens192": {
   "addresses": [
    "x.x.x.27/20"  <--- BUILD IP
   ],
   "dhcp4": false,
   "gateway4": "x.x.x.1",
   "nameservers": {
    "addresses": [
     "x.x.x.x",
     "x.x.x.x"
    ]
   }
  }
 },
 "version": 2
}

cloud-user@vm:~$ sudo cat /etc/netplan/50-cloud-init.yaml 
# This file is generated from information provided by the datasource.  Changes
# to it will not persist across an instance reboot.  To disable cloud-init's
# network configuration capabilities, write a file
# /etc/cloud/cloud.cfg.d/99-disable-network-config.cfg with the following:
# network: {config: disabled}
network:
    ethernets:
        ens192:
            addresses:
            - x.x.x.27/20 <--- BUILD IP
            dhcp4: false
            gateway4: x.x.x.1
            nameservers:
                addresses:
                -  x.x.x.x
                -  x.x.x.x
    version: 2

cloud-user@vm:~$ sudo cat /etc/cloud/cloud.cfg.d/90-installer-network.cfg 
# This is the network config written by 'subiquity'
network:
  ethernets:
    ens192:
      addresses:
      - x.x.x.27/20 <--- BUILD IP
      dhcp4: false
      gateway4: x.x.x.1
      nameservers:
        addresses:
        - x.x.x.x
        - x.x.x.x
  version: 2



cloud-user@anthony:~$ sudo cat /run/cloud-init/instance-data.json 
{
 "base64_encoded_keys": [],
 "ds": {
  "_doc": "EXPERIMENTAL: The structure and format of content scoped under the 'ds' key may change in subsequent releases of cloud-init.",
  "meta_data": {
   "hostname": "vm",
   "instance-id": "156d2742-3d2b-4b18-fb77-9419a4afb558",
   "local-hostname": "vm.xxxx.xxx",
   "local-ipv4": "x.x.x.27", <--- BUILD IP
   "local_hostname": "vm",
   "network": {
    "config": {
     "ethernets": {
      "ens192": {
       "addresses": [
        "x.x.x.24/20" <--- GOOD IP FROM CLOUD INIT
       ],
       "dhcp4": false,
       "gateway4": "x.x.x.1",
       "nameservers": {
        "addresses": [
         "x.x.x.x",
         "x.x.x.x"
        ]
       }
      }
     },
     "version": 2
    },
    "interfaces": {
     "by-ipv4": {
      "x.x.x.27": { <--- BUILD IP
       "broadcast": "x.x.x.255",
       "mac": "00:50:56:a7:88:33",
       "netmask": "255.255.240.0"
      }
     },
     "by-ipv6": {},
     "by-mac": {
      "00:50:56:a7:88:33": {
       "ipv4": [
        {
         "addr": "x.x.x.27", <--- BUILD IP
         "broadcast": "x.x.x.255",
         "netmask": "255.255.240.0"
        }
       ],
       "ipv6": []
      }
     }
    }
   }
  }
 },



```

**Comment by tenthirtyam** [comment]
Duplicate #979

**Comment by sticky-note** [comment]
I ran into the same problem today, it seems to come from `GRUB_CMDLINE_LINUX_DEFAULT` being set on `packer` template provisioning via `cloud-init` installer.
It is set by packer `boot_command` on kernel cmdline arguments and tell `cloud-init` to use a specific DataSource (`nocloud-net` or `nocloud`) and the installer persists these arguments on grub configuration in `/etc/default/grub`.
It it the reason why `cloud-init` does not read `VMware` DataSource.

I ended up with this https://github.com/vmware-samples/packer-examples-for-vsphere/pull/982 to restablish an empty `GRUB_CMDLINE_LINUX_DEFAULT` as `late_command`

It works on my side
Hope it'll help someone.
